crates/core/src/database_instance_context_controller.rs

```diff
@@ -160,14 +163,43 @@ impl DatabaseInstanceContextController {
         loop {
             let tick = interval.tick().await;
             let dt = tick - prev_tick;
-            for (cell, prev_disk_usage) in self.contexts.lock().unwrap().values_mut() {
+
+            let Some(this) = this.upgrade() else { break };
+
+            for (cell, prev_disk_usage) in this.contexts.lock().unwrap().values_mut() {
                 if let Some((db, _)) = cell.get() {
                     let disk_usage = db.total_disk_usage().or(*prev_disk_usage);
-                    self.energy_monitor
+                    this.energy_monitor
                         .record_disk_usage(&db.database, db.database_instance_id, disk_usage.sum(), dt);
                     *prev_disk_usage = disk_usage;
                 }
             }
+
+            prev_tick = tick;
+        }
+    }
+
+    const MEM_METERING_INTERVAL: Duration = Duration::from_secs(100);
```

Comment:
Why not use the same duration as disk metering?

Reply:
lmao, I thought Tyler said that the interval should be 100, not that the energy multiplier should be 100.